metrics/src/metrics.rs

```diff
@@ -159,25 +159,12 @@ impl MetricsAgent {
         Self { sender }
     }
 
-    fn collect_points(
-        points_map: &mut HashMap<log::Level, (CounterMap, Vec<DataPoint>)>,
-    ) -> Vec<DataPoint> {
-        let points: Vec<DataPoint> = [
-            Level::Error,
-            Level::Warn,
-            Level::Info,
-            Level::Debug,
-            Level::Trace,
-        ]
-        .iter()
-        .filter_map(|level| points_map.remove(level))
-        .flat_map(|(counters, points)| {
-            let counter_points = counters.into_iter().map(|(_, v)| v.into());
-            points.into_iter().chain(counter_points)
-        })
-        .collect();
-        points_map.clear();
-        points
+    fn collect_points(points: &mut Vec<DataPoint>, counters: &mut CounterMap) -> Vec<DataPoint> {
+        let mut ret: Vec<DataPoint> = points.drain(..).collect();
```

Comment:
maybe:
```
let mut empty = Vec::default();
std::mem::swap(&mut empty, points);
```

the result is what was in 'points' is now in 'empty' and vice-a-versa with no copies or reallocations of anything.

Reply:
Updated. And this further improves the performance. 😉 
```
test bench_datapoint_submission ... bench:     458,256 ns/iter (+/- 40,932)
```